### PR TITLE
Upgrade ks-installer chart to v3.2.0

### PR DESCRIPTION
**Pull request by mangoGoForward**
### What this PR dose?
Upgrade ks-installer chart to v3.2.0

### What this PR does / why we need it:
please see #185.

### Which issue(s) this PR fixes:

Fixes #185.

### CODE REVIEW

**Comment by pixiake** [comment]
Many configurations in the `clusterConfiguration` have changed.
https://github.com/kubesphere/ks-installer/blob/release-3.2/deploy/cluster-configuration.yaml

I think everything in the `spec` of  `clusterConfiguration` can be defined as configurable parameter,  so we don't have to consider the configuration changes in it.

**Comment by mangoGoForward** [comment]
> Many configurations in the `clusterConfiguration` have changed. https://github.com/kubesphere/ks-installer/blob/release-3.2/deploy/cluster-configuration.yaml
> 
> I think everything in the `spec` of `clusterConfiguration` can be defined as a configurable parameter, so we don't have to consider the configuration changes in it.

Do you mean use one parameter `spec` instead of support detailed configuration items? and do we need verify the `spec` parameter?

**Comment by pixiake** [comment]
> > Many configurations in the `clusterConfiguration` have changed. https://github.com/kubesphere/ks-installer/blob/release-3.2/deploy/cluster-configuration.yaml
> > I think everything in the `spec` of `clusterConfiguration` can be defined as a configurable parameter, so we don't have to consider the configuration changes in it.
> 
> Do you mean use one parameter `spec` instead of support detailed configuration items? and do we need verify the `spec` parameter?

Yes
The configuration of ks-installer is not a fixed structure, and there are many configuration items, it is not realistic to list each item separately for configuration, so I think it is more flexible to specify the configuration directly with one parameter, such as `clusterConfiguration`.  
How do you like it ? 
```
replicaCount: 1

...

clusterConfiguration:
  persistence:
    storageClass: ""
  authentication:
    jwtSecret: ""  
  local_registry: ""    
  etcd:
    monitoring: false   
    endpointIps: localhost 
    port: 2379  
    tlsEnable: true
  common:
    core:
      console:
        enableMultiLogin: true 
        port: 30880
        type: NodePort
    redis:
      enabled: false
      volumeSize: 2Gi # Redis PVC size.
    openldap:
      enabled: false
      volumeSize: 2Gi   # openldap PVC size.
    minio:
      volumeSize: 20Gi # Minio PVC size.
    monitoring:
      # type: external   # Whether to specify the external prometheus stack, and need to modify the endpoint at the next line.
      endpoint: http://prometheus-operated.kubesphere-monitoring-system.svc:9090 # Prometheus endpoint to get metrics data.
      GPUMonitoring:     # Enable or disable the GPU-related metrics. If you enable this switch but have no GPU resources, Kubesphere will set it to zero. 
        enabled: false
    gpu:                 # Install GPUKinds. The default GPU kind is nvidia.com/gpu. Other GPU kinds can be added here according to your needs. 
      kinds:         
      - resourceName: "nvidia.com/gpu"
        resourceType: "GPU"
        default: true
    es:   # Storage backend for logging, events and auditing.
      # master:
      #   volumeSize: 4Gi  # The volume size of Elasticsearch master nodes.
      #   replicas: 1      # The total number of master nodes. Even numbers are not allowed.
      #   resources: {}
      # data:
      #   volumeSize: 20Gi  # The volume size of Elasticsearch data nodes.
      #   replicas: 1       # The total number of data nodes.
      #   resources: {}
      logMaxAge: 7             # Log retention time in built-in Elasticsearch. It is 7 days by default.
      elkPrefix: logstash      # The string making up index names. The index name will be formatted as ks-<elk_prefix>-log.
      basicAuth:
        enabled: false
        username: ""
        password: ""
      externalElasticsearchHost: ""
      externalElasticsearchPort: ""
  alerting:                # (CPU: 0.1 Core, Memory: 100 MiB) It enables users to customize alerting policies to send messages to receivers in time with different time intervals and alerting levels to choose from.
    enabled: false         # Enable or disable the KubeSphere Alerting System.
    # thanosruler:
    #   replicas: 1
    #   resources: {}
  auditing:                # Provide a security-relevant chronological set of records，recording the sequence of activities happening on the platform, initiated by different tenants.
    enabled: false         # Enable or disable the KubeSphere Auditing Log System.
    # operator:
    #   resources: {}
    # webhook:
    #   resources: {}
```

**Comment by mangoGoForward** [comment]
> > > Many configurations in the `clusterConfiguration` have changed. https://github.com/kubesphere/ks-installer/blob/release-3.2/deploy/cluster-configuration.yaml
> > > I think everything in the `spec` of `clusterConfiguration` can be defined as a configurable parameter, so we don't have to consider the configuration changes in it.
> > 
> > 
> > Do you mean use one parameter `spec` instead of support detailed configuration items? and do we need verify the `spec` parameter?
> 
> Yes The configuration of ks-installer is not a fixed structure, and there are many configuration items, it is not realistic to list each item separately for configuration, so I think it is more flexible to specify the configuration directly with one parameter, such as `clusterConfiguration`. How do you like it ?
> 
> ```
> replicaCount: 1
> 
> ...
> 
> clusterConfiguration:
>   persistence:
>     storageClass: ""
>   authentication:
>     jwtSecret: ""  
>   local_registry: ""    
>   etcd:
>     monitoring: false   
>     endpointIps: localhost 
>     port: 2379  
>     tlsEnable: true
>   common:
>     core:
>       console:
>         enableMultiLogin: true 
>         port: 30880
>         type: NodePort
>     redis:
>       enabled: false
>       volumeSize: 2Gi # Redis PVC size.
>     openldap:
>       enabled: false
>       volumeSize: 2Gi   # openldap PVC size.
>     minio:
>       volumeSize: 20Gi # Minio PVC size.
>     monitoring:
>       # type: external   # Whether to specify the external prometheus stack, and need to modify the endpoint at the next line.
>       endpoint: http://prometheus-operated.kubesphere-monitoring-system.svc:9090 # Prometheus endpoint to get metrics data.
>       GPUMonitoring:     # Enable or disable the GPU-related metrics. If you enable this switch but have no GPU resources, Kubesphere will set it to zero. 
>         enabled: false
>     gpu:                 # Install GPUKinds. The default GPU kind is nvidia.com/gpu. Other GPU kinds can be added here according to your needs. 
>       kinds:         
>       - resourceName: "nvidia.com/gpu"
>         resourceType: "GPU"
>         default: true
>     es:   # Storage backend for logging, events and auditing.
>       # master:
>       #   volumeSize: 4Gi  # The volume size of Elasticsearch master nodes.
>       #   replicas: 1      # The total number of master nodes. Even numbers are not allowed.
>       #   resources: {}
>       # data:
>       #   volumeSize: 20Gi  # The volume size of Elasticsearch data nodes.
>       #   replicas: 1       # The total number of data nodes.
>       #   resources: {}
>       logMaxAge: 7             # Log retention time in built-in Elasticsearch. It is 7 days by default.
>       elkPrefix: logstash      # The string making up index names. The index name will be formatted as ks-<elk_prefix>-log.
>       basicAuth:
>         enabled: false
>         username: ""
>         password: ""
>       externalElasticsearchHost: ""
>       externalElasticsearchPort: ""
>   alerting:                # (CPU: 0.1 Core, Memory: 100 MiB) It enables users to customize alerting policies to send messages to receivers in time with different time intervals and alerting levels to choose from.
>     enabled: false         # Enable or disable the KubeSphere Alerting System.
>     # thanosruler:
>     #   replicas: 1
>     #   resources: {}
>   auditing:                # Provide a security-relevant chronological set of records，recording the sequence of activities happening on the platform, initiated by different tenants.
>     enabled: false         # Enable or disable the KubeSphere Auditing Log System.
>     # operator:
>     #   resources: {}
>     # webhook:
>     #   resources: {}
> ```

It may let user confused about the structure of cluster configuration.

**Comment by pixiake** [comment]
> > > > Many configurations in the `clusterConfiguration` have changed. https://github.com/kubesphere/ks-installer/blob/release-3.2/deploy/cluster-configuration.yaml
> > > > I think everything in the `spec` of `clusterConfiguration` can be defined as a configurable parameter, so we don't have to consider the configuration changes in it.
> > > 
> > > 
> > > Do you mean use one parameter `spec` instead of support detailed configuration items? and do we need verify the `spec` parameter?
> > 
> > 
> > Yes The configuration of ks-installer is not a fixed structure, and there are many configuration items, it is not realistic to list each item separately for configuration, so I think it is more flexible to specify the configuration directly with one parameter, such as `clusterConfiguration`. How do you like it ?
> > ```
> > replicaCount: 1
> > 
> > ...
> > 
> > clusterConfiguration:
> >   persistence:
> >     storageClass: ""
> >   authentication:
> >     jwtSecret: ""  
> >   local_registry: ""    
> >   etcd:
> >     monitoring: false   
> >     endpointIps: localhost 
> >     port: 2379  
> >     tlsEnable: true
> >   common:
> >     core:
> >       console:
> >         enableMultiLogin: true 
> >         port: 30880
> >         type: NodePort
> >     redis:
> >       enabled: false
> >       volumeSize: 2Gi # Redis PVC size.
> >     openldap:
> >       enabled: false
> >       volumeSize: 2Gi   # openldap PVC size.
> >     minio:
> >       volumeSize: 20Gi # Minio PVC size.
> >     monitoring:
> >       # type: external   # Whether to specify the external prometheus stack, and need to modify the endpoint at the next line.
> >       endpoint: http://prometheus-operated.kubesphere-monitoring-system.svc:9090 # Prometheus endpoint to get metrics data.
> >       GPUMonitoring:     # Enable or disable the GPU-related metrics. If you enable this switch but have no GPU resources, Kubesphere will set it to zero. 
> >         enabled: false
> >     gpu:                 # Install GPUKinds. The default GPU kind is nvidia.com/gpu. Other GPU kinds can be added here according to your needs. 
> >       kinds:         
> >       - resourceName: "nvidia.com/gpu"
> >         resourceType: "GPU"
> >         default: true
> >     es:   # Storage backend for logging, events and auditing.
> >       # master:
> >       #   volumeSize: 4Gi  # The volume size of Elasticsearch master nodes.
> >       #   replicas: 1      # The total number of master nodes. Even numbers are not allowed.
> >       #   resources: {}
> >       # data:
> >       #   volumeSize: 20Gi  # The volume size of Elasticsearch data nodes.
> >       #   replicas: 1       # The total number of data nodes.
> >       #   resources: {}
> >       logMaxAge: 7             # Log retention time in built-in Elasticsearch. It is 7 days by default.
> >       elkPrefix: logstash      # The string making up index names. The index name will be formatted as ks-<elk_prefix>-log.
> >       basicAuth:
> >         enabled: false
> >         username: ""
> >         password: ""
> >       externalElasticsearchHost: ""
> >       externalElasticsearchPort: ""
> >   alerting:                # (CPU: 0.1 Core, Memory: 100 MiB) It enables users to customize alerting policies to send messages to receivers in time with different time intervals and alerting levels to choose from.
> >     enabled: false         # Enable or disable the KubeSphere Alerting System.
> >     # thanosruler:
> >     #   replicas: 1
> >     #   resources: {}
> >   auditing:                # Provide a security-relevant chronological set of records，recording the sequence of activities happening on the platform, initiated by different tenants.
> >     enabled: false         # Enable or disable the KubeSphere Auditing Log System.
> >     # operator:
> >     #   resources: {}
> >     # webhook:
> >     #   resources: {}
> > ```
> 
> It may let user confused about the structure of cluster configuration.



Otherwise, [the template](https://github.com/kubesphere/helm-charts/blob/master/src/main/ks-installer/templates/clusterconfiguration.yaml) should be modified according to [the new configuration](https://github.com/kubesphere/ks-installer/blob/master/deploy/cluster-configuration.yaml). Or do you have any better ideas ?

**Comment by mangoGoForward** [comment]
> > > > > Many configurations in the `clusterConfiguration` have changed. https://github.com/kubesphere/ks-installer/blob/release-3.2/deploy/cluster-configuration.yaml
> > > > > I think everything in the `spec` of `clusterConfiguration` can be defined as a configurable parameter, so we don't have to consider the configuration changes in it.
> > > > 
> > > > 
> > > > Do you mean use one parameter `spec` instead of support detailed configuration items? and do we need verify the `spec` parameter?
> > > 
> > > 
> > > Yes The configuration of ks-installer is not a fixed structure, and there are many configuration items, it is not realistic to list each item separately for configuration, so I think it is more flexible to specify the configuration directly with one parameter, such as `clusterConfiguration`. How do you like it ?
> > > ```
> > > replicaCount: 1
> > > 
> > > ...
> > > 
> > > clusterConfiguration:
> > >   persistence:
> > >     storageClass: ""
> > >   authentication:
> > >     jwtSecret: ""  
> > >   local_registry: ""    
> > >   etcd:
> > >     monitoring: false   
> > >     endpointIps: localhost 
> > >     port: 2379  
> > >     tlsEnable: true
> > >   common:
> > >     core:
> > >       console:
> > >         enableMultiLogin: true 
> > >         port: 30880
> > >         type: NodePort
> > >     redis:
> > >       enabled: false
> > >       volumeSize: 2Gi # Redis PVC size.
> > >     openldap:
> > >       enabled: false
> > >       volumeSize: 2Gi   # openldap PVC size.
> > >     minio:
> > >       volumeSize: 20Gi # Minio PVC size.
> > >     monitoring:
> > >       # type: external   # Whether to specify the external prometheus stack, and need to modify the endpoint at the next line.
> > >       endpoint: http://prometheus-operated.kubesphere-monitoring-system.svc:9090 # Prometheus endpoint to get metrics data.
> > >       GPUMonitoring:     # Enable or disable the GPU-related metrics. If you enable this switch but have no GPU resources, Kubesphere will set it to zero. 
> > >         enabled: false
> > >     gpu:                 # Install GPUKinds. The default GPU kind is nvidia.com/gpu. Other GPU kinds can be added here according to your needs. 
> > >       kinds:         
> > >       - resourceName: "nvidia.com/gpu"
> > >         resourceType: "GPU"
> > >         default: true
> > >     es:   # Storage backend for logging, events and auditing.
> > >       # master:
> > >       #   volumeSize: 4Gi  # The volume size of Elasticsearch master nodes.
> > >       #   replicas: 1      # The total number of master nodes. Even numbers are not allowed.
> > >       #   resources: {}
> > >       # data:
> > >       #   volumeSize: 20Gi  # The volume size of Elasticsearch data nodes.
> > >       #   replicas: 1       # The total number of data nodes.
> > >       #   resources: {}
> > >       logMaxAge: 7             # Log retention time in built-in Elasticsearch. It is 7 days by default.
> > >       elkPrefix: logstash      # The string making up index names. The index name will be formatted as ks-<elk_prefix>-log.
> > >       basicAuth:
> > >         enabled: false
> > >         username: ""
> > >         password: ""
> > >       externalElasticsearchHost: ""
> > >       externalElasticsearchPort: ""
> > >   alerting:                # (CPU: 0.1 Core, Memory: 100 MiB) It enables users to customize alerting policies to send messages to receivers in time with different time intervals and alerting levels to choose from.
> > >     enabled: false         # Enable or disable the KubeSphere Alerting System.
> > >     # thanosruler:
> > >     #   replicas: 1
> > >     #   resources: {}
> > >   auditing:                # Provide a security-relevant chronological set of records，recording the sequence of activities happening on the platform, initiated by different tenants.
> > >     enabled: false         # Enable or disable the KubeSphere Auditing Log System.
> > >     # operator:
> > >     #   resources: {}
> > >     # webhook:
> > >     #   resources: {}
> > > ```
> > 
> > 
> > It may let user confused about the structure of cluster configuration.
> 
> Otherwise, [the template](https://github.com/kubesphere/helm-charts/blob/master/src/main/ks-installer/templates/clusterconfiguration.yaml) should be modified according to [the new configuration](https://github.com/kubesphere/ks-installer/blob/master/deploy/cluster-configuration.yaml). Or do you have any better ideas ?
I have no idea right now, your suggestion solved synchronization problem with the new cluster configuration in ks-installer, I will refactor code firstly, and if have other great ideas, we can deal with it again.

**Comment by mangoGoForward** [comment]
Hi, I have refactored cc template of ks-installer, could you help me to provide some suggestions? @pixiake

**Comment by pixiake** [comment]
> Hi, I have refactored cc template of ks-installer, could you help me to provide some suggestions? @pixiake

https://github.com/kubesphere/helm-charts/blob/c499885f56db53f77e438ac0904987849bb01a77/src/main/ks-installer/templates/NOTES.txt#L2

Thanks，I think this place should be changed to `cc.common.core.console.port` .

**Comment by mangoGoForward** [comment]
> > Hi, I have refactored cc template of ks-installer, could you help me to provide some suggestions? @pixiake
> 
> https://github.com/kubesphere/helm-charts/blob/c499885f56db53f77e438ac0904987849bb01a77/src/main/ks-installer/templates/NOTES.txt#L2
> 
> Thanks，I think this place should be changed to `cc.common.core.console.port` .

All right, Done.

**Comment by zheng1** [comment]
@mangoGoForward please update the version in Chart.yaml

**Comment by mangoGoForward** [comment]
> @mangoGoForward please update the version in Chart.yaml

Done.

File: src/main/ks-installer/Chart.yaml [hunk 1]
```diff
@@ -18,9 +18,9 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.2.2
+version: 1.0.0
```
Comment (pixiake) [inline src/main/ks-installer/Chart.yaml:21]: I think it might be better to change the chart version to `0.3.0`.

Comment (mangoGoForward) [reply src/main/ks-installer/Chart.yaml:21]: But this update has changed the Values structure which is incompatible with old version.

Comment (mangoGoForward) [reply src/main/ks-installer/Chart.yaml]: Done.

**Comment by pixiake** [comment]
@zheng1  I think this pr can be merged，and very grateful to @mangoGoForward  for his contribution.

**Pull request merged**